### ISSUE TITLE
Issue during containerized deployment mentions "possible bug", but I'm not so sure.

### ISSUE THREAD

**Issue by anthonysiano** [openshift/openshift-ansible#8950]
#### Description

> error running prerequisites playbook using the atomic install during openshift_sanitize_inventory.

##### Version

Whatever ansible version is in the atomic container, I don't know how to check.

##### Steps To Reproduce
1. atomic install --system --storage=ostree --set INVENTORY_FILE=/opt/osc-dev-build/inventory.yml --set PLAYBOOK_FILE=/usr/share/ansible/openshift-ansible/playbooks/prerequisites.yml --set OPTS="-vvv" docker.io/openshift/origin-ansible:v3.9

##### Observed Results
```
TASK [openshift_sanitize_inventory : conditional_set_fact] *********************
task path: /usr/share/ansible/openshift-ansible/roles/openshift_sanitize_inventory/tasks/__deprecations_logging.yml:5
Saturday 23 June 2018  14:01:30 +0000 (0:00:00.348)       0:00:04.312 *********
ERROR! Unexpected Exception, this is probably a bug: update expected at most 1 arguments, got 2
the full traceback was:

Traceback (most recent call last):
  File "/usr/bin/ansible-playbook", line 118, in <module>
    exit_code = cli.run()
  File "/usr/lib/python2.7/site-packages/ansible/cli/playbook.py", line 122, in run
    results = pbex.run()
  File "/usr/lib/python2.7/site-packages/ansible/executor/playbook_executor.py", line 159, in run
    result = self._tqm.run(play=play)
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_queue_manager.py", line 289, in run
    play_return = strategy.run(iterator, play_context)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/strategy/linear.py", line 313, in run
    results += self._process_pending_results(iterator, max_passes=max(1, int(len(self._tqm._workers) * 0.1)))
  File "/usr/lib/python2.7/site-packages/ansible/plugins/strategy/__init__.py", line 117, in inner
    results = func(self, iterator, one_pass=one_pass, max_passes=max_passes)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/strategy/__init__.py", line 606, in _process_pending_results
    self._variable_manager.set_host_facts(target_host, result_item['ansible_facts'].copy())
  File "/usr/lib/python2.7/site-packages/ansible/vars/manager.py", line 587, in set_host_facts
    self._fact_cache.update(host.name, facts)
TypeError: update expected at most 1 arguments, got 2
```

##### Additional Information
* Red Hat Enterprise Linux Server release 7.5 (Maipo) host VMs running on vSphere 6.5
* Inventory File (part):
```
[OSEv3:children]
masters
nodes
etcd
lb

# Set variables common for all OSEv3 hosts
[OSEv3:vars]
ansible_ssh_user=root
openshift_deployment_type=origin
os_firewall_use_firewalld=True
openshift_hosted_registry_cert_expire_days=1825
openshift_ca_cert_expire_days=1825
openshift_node_cert_expire_days=1825
openshift_master_cert_expire_days=1825
etcd_ca_default_days=1825

# Uncomment the following to enable htpasswd authentication; defaults to
# DenyAllPasswordIdentityProvider.
#openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider'}]

# Native high availability cluster method with optional load balancer.
# If no lb group is defined installer assumes that a load balancer has
# been preconfigured. For installation the value of
# openshift_master_cluster_hostname must resolve to the load balancer
# or to one or all of the masters defined in the inventory if no load
# balancer is present.
openshift_master_cluster_method=native
```

**Comment by vrutkovs** [comment]
This looks like an ansible issue, is this transient or reproducible all the time in your case?

**Comment by anthonysiano** [comment]
Not sure, I ended up switching the RPM method instead and it gets past this part and prerequisites actually completed so now I'm further along tracking down another issue with the deploy_cluster.yml instead.

**Comment by olsopon** [comment]
I can reproduce this every time (cannot find a work around). Python version is 2.7.5 for me.
set_host_facts seem broken

    def set_host_facts(self, host, facts):
        '''
        Sets or updates the given facts for a host in the fact cache.
        '''

        if not isinstance(facts, dict):
            raise AnsibleAssertionError("the type of 'facts' to set for host_facts should be a dict but is a %s" % type(facts))

        if host.name not in self._fact_cache:
            self._fact_cache[host.name] = facts
        else:
            try:
                self._fact_cache.update(host.name, facts)
            except KeyError:
                self._fact_cache[host.name] = facts

**Comment by vrutkovs** [comment]
Closing as this as the issue is about 3.9 and has been resolved. If you encounter a similar issue, please open a new ticket and specify required info

**Comment by mriedmann** [comment]
I hit the same problem during a v3.11.0 deployment but only after enabling the jsonfile fact_caching plugin. 

```
python: 2.7.5
ansible: 2.6.6
```

So a workaround for me was to swith to the default (memory) fact_caching plugin.
